core/src/replicator.rs

```diff
@@ -599,6 +600,126 @@ impl Replicator {
             "Couldn't get blockhash or slot",
         ))?
     }
+
+    /// Ask a replicator to populate a given blocktree with its segment.
+    /// Return the slot at the start of the replicator's segment
+    ///
+    /// It is recommended to use a temporary blocktree for this since the download will not verify
+    /// blobs received and might impact the chaining of blobs across slots
+    pub fn download_from_replicator(
+        cluster_info: &Arc<RwLock<ClusterInfo>>,
+        replicator_info: &ContactInfo,
+        blocktree: &Arc<Blocktree>,
+    ) -> Result<(u64)> {
+        // Create a client which downloads from the replicator and see that it
+        // can respond with blobs.
+        let start_slot = Self::get_replicator_segment_slot(replicator_info.storage_addr);
+        info!("Replicator download: start at {}", start_slot);
+
+        let exit = Arc::new(AtomicBool::new(false));
+        let (s_reader, r_reader) = channel();
+        let repair_socket = Arc::new(UdpSocket::bind("0.0.0.0:0").unwrap());
+        let t_receiver = blob_receiver(repair_socket.clone(), &exit, s_reader);
+        let id = cluster_info.read().unwrap().id();
+        info!(
+            "Sending repair requests from: {} to: {}",
+            cluster_info.read().unwrap().my_data().id,
+            replicator_info.gossip
+        );
+        let repair_slot_range = RepairSlotRange {
+            start: start_slot,
+            end: start_slot + SLOTS_PER_SEGMENT,
+        };
+        // try for 60 seconds //TODO needs tuning if segments are huge
+        for _ in 0..120 {
```

Comment:
it's actually 1.5s * 120 right?

Reply:
yeah you're right, that is the worst case perf. hm.. should I fix it? I am indifferent...

Reply:
I think just change the comment to kind of match what it is doing. Whether it is 60 seconds or 180 seconds doesn't matter.

Reply:
Yeah fair enough I'll fix in a sec